### PR TITLE
chore: simplify config file generation

### PR DESCRIPTION
**Pull request by kayra1**
# Description

This change removes the config yaml file from the charm repo and generates it live instead.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -147,10 +148,18 @@ def _configure_notary_config_file(self):
...
             logger.info("Config file already created.")
         except ops.pebble.PathError:
-            config_file = open("src/config/config.yaml").read()
             self.container.make_dir(path=f"{WORKLOAD_CONFIG_PATH}/config", make_parents=True)
             self.container.push(
-                path=f"{WORKLOAD_CONFIG_PATH}/config/config.yaml", source=config_file
+                path=f"{WORKLOAD_CONFIG_PATH}/config/config.yaml",
+                source=yaml.dump(
+                    data={
+                        "key_path": "/etc/notary/config/private_key.pem",
```
Comment (gruyaume) [inline src/charm.py:156]: Let's use our `WORKLOAD_CONFIG_PATH` variable here and in cert path

File: src/charm.py [hunk 2]
```diff
@@ -147,10 +148,18 @@ def _configure_notary_config_file(self):
...
+                        "cert_path": "/etc/notary/config/certificate.pem",
+                        "db_path": "/var/lib/notary/database/certs.db",
```
Comment (gruyaume) [inline src/charm.py:158]: We could probably use a variable for the db path too. Even though it's only used here, it's associated to a specific place in the workload container which has a mount

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: src/charm.py (hunk 1 above)
Comment (kayra1) [reply src/charm.py]: done

File: src/charm.py (hunk 2 above)
Comment (kayra1) [reply src/charm.py]: agreed, done

**Pull request merged**